### PR TITLE
(CONT-359) Syntax update

### PR DESCRIPTION
**Pull request by LukasAud**
Code now compliant with rules regarding:

legacy facts
parameter datatypes
parameter documentation
top-scope facts

### CODE REVIEW

File: manifests/params.pp [hunk 1]
```diff
@@ -340,7 +340,7 @@
       $config_file         = '/etc/my.cnf'
       $includedir          = undef
       $datadir             = '/var/mysql'
-      $log_error           = "/var/mysql/${::hostname}.err"
+      $log_error           = "/var/mysql/${facts['os']['hostname']}.err"
```
Comment (smortex) [inline manifests/params.pp:343]:
```suggestion
      $log_error           = "/var/mysql/${facts['networking']['hostname']}.err"
```

**Review by smortex — COMMENTED** [review]
:wave: it seems like I had a few more legacy facts removed in `/spec/` in #1535

File: manifests/params.pp (hunk 1 above)
Comment (LukasAud) [reply manifests/params.pp:343]: Investigated this and confirmed a number of facts hidden in the test files that were not caught by our Lint plugin. Will amend soon.

Comment (LukasAud) [reply manifests/params.pp]: That should be those addressed

**Review by smortex — APPROVED** [review]
Looks good!

**Review by david22swan — APPROVED** [review]
LGTM

**Comment by jdyb** [comment]
Was it intentional to change `Mysql::Server` parameter `package_ensure` to no longer allow `latest` ?
It used to work for us and when we updated the module we hit the type restriction.

**Pull request merged**